### DISCUSSION TITLE
refusing to allow a GitHub App to create or update workflow without workflows permission

### DISCUSSION THREAD

**Discussion by Xuanwo** [Mergifyio/mergify#5061] [Q&A]
Hi, we met the following problems every time we have updated our workflow.

Like in PR https://github.com/datafuselabs/databend/pull/9147

```
refusing to allow a GitHub App to create or update workflow .github/workflows/databend-release.yml without workflows permission
err-code: 704E2
```

![image](https://user-images.githubusercontent.com/5351546/206631138-8583c897-a8b1-4b65-82d6-f8ab8edc7b71.png)

But we are sure that we have given `workflows` permission to mergify.

Our workflows are listed here: https://github.com/datafuselabs/databend/tree/main/.github/workflows

**Comment by jd** [comment]
You need to update the permission granted to Mergify. An admin should log into their Mergify dashboard and ensure permissions are up to date.

**Reply by Xuanwo** [reply]
I'm not sure what's the permissions mentioned here. But it looks good from github side:

![image](https://user-images.githubusercontent.com/5351546/206671158-5827a19f-1979-4025-b0d8-8151ddadc032.png)

**Reply by jd** [reply]
This does not show permissions. Did you check from your Mergify dashboard if you had new permission to accept?

**Reply by Xuanwo** [reply]
![image](https://user-images.githubusercontent.com/5351546/206696874-8c8bde97-e744-4334-9b67-f68a4bc3a1ca.png)

No notification from the dashboard.

**Comment by Xuanwo** [comment]
databend.release.yml has permission settings like:

```yaml
permissions:
  id-token: write
  contents: write
```

Does this matters?

**Reply by jd** [reply]
Not at all.

**Comment by AlekSi** [comment]
> refusing to allow a GitHub App to create or update workflow .github/workflows/website-preview.yml without workflows permission
> err-code: 8D91D

* https://github.com/FerretDB/FerretDB/pull/2422/checks?check_run_id=12834081044

Mergify has access to:
> Read and write access to checks, code, issues, pages, pull requests, and workflows

No prompt in the dashboard.
